### PR TITLE
qemu-arm-virt: Add support for 32bit arm hyp mode configurations

### PR DESCRIPTION
**Pull request by kent-mcleod**
Add support for the qemu-arm-virt simulation platform to run seL4 in hyp mode.

### CODE REVIEW

File: include/arch/arm/arch/32/mode/hardware.h [hunk 1]
```diff
@@ -85,6 +85,10 @@
 #ifndef __ASSEMBLER__
 /* It is required that USER_TOP must be aligned to at least 20 bits */
 compile_assert(USER_TOP_correctly_aligned, IS_ALIGNED(USER_TOP, 20));
+/* It is required on arm_hyp that USER_TOP isn't lower than the top GiB */
```
Comment (Indanz) [inline include/arch/arm/arch/32/mode/hardware.h:88]: Where does this requirement come from?

File: src/plat/qemu-arm-virt/config.cmake [hunk 2]
```diff
@@ -60,7 +71,7 @@ if(KernelPlatformQEMUArmVirt)
     config_set(KernelARMPlatform ARM_PLAT qemu-arm-virt)
     set(DTBPath "${CMAKE_BINARY_DIR}/virt.dtb")
     set(DTSPath "${CMAKE_BINARY_DIR}/virt.dts")
-    if(KernelArmHypervisorSupport)
+    if(KernelArmHypervisorSupport OR KernelSel4ArchArmHyp)
```
Comment (Indanz) [inline src/plat/qemu-arm-virt/config.cmake:74]: Why have both? Is there a difference in meaning between the two?

**Review by Indanz — APPROVED** [review]
Looks good to me.

File: include/arch/arm/arch/32/mode/hardware.h (hunk 1 above)
Comment (kent-mcleod) [reply include/arch/arm/arch/32/mode/hardware.h:88]: https://github.com/seL4/seL4/blob/master/src/arch/arm/32/kernel/vspace.c#L316.  The kernel has it's own PL2 address space, but only uses the top 1 GiB.

File: src/plat/qemu-arm-virt/config.cmake (hunk 2 above)
Comment (kent-mcleod) [reply src/plat/qemu-arm-virt/config.cmake:74]: `KernelArmHypervisorSupport` may not be set at this point as it's not declared until a later CMake file is processed.

**Review by yyshen — APPROVED** [review]
Good work!

**Pull request merged**